modules/examples/sources/msg-generator/msg-generator-source-options.h

```diff
@@ -26,20 +26,24 @@
 
 #include "logsource.h"
 #include "template/templates.h"
+#include "logmsg/logmsg.h"
+#include <gmodule.h>
 
 typedef struct
 {
   LogSourceOptions super;
   guint freq;
   gint max_num;
   LogTemplate *template;
+  GHashTable *name_value;
 } MsgGeneratorSourceOptions;
 
 static inline void
 msg_generator_source_options_defaults(MsgGeneratorSourceOptions *self)
 {
   log_source_options_defaults(&self->super);
   self->freq = 1000;
+  self->name_value = g_hash_table_new(g_str_hash, g_str_equal);
```

Comment:
Please free the HashTable, when we don't need it anymore.
Also `g_hash_table_new_fullI()` should be used here, so the keys and values are freed as well.

Reply:
One other thing, I think creating the HashTable should be in `msg_generator_sd_new()`. I know that `msg_generator_source_options_defaults()` is only called there, so the result is the same, but it is an allocation/initialization step, and not a default value setting step.